rhc-ose-ansible/roles/docker/tasks/main.yml

```diff
@@ -0,0 +1,45 @@
+---
+- name: "Setting Docker Facts"
+  set_fact:
+    docker_storage_block_device: "{{ docker_storage_block_device | default(default_docker_storage_block_device) }}"
+    docker_storage_volume_group: "{{ docker_storage_volume_group | default(default_docker_storage_volume_group) }}"
+
+- name: "Install Docker"
+  yum: 
+    name: docker
+    state: latest
+  notify:
+    - enable docker
+
+- name: "Confige Docker"
+  lineinfile: 
+    dest: /etc/sysconfig/docker 
+    regexp: '^OPTIONS=.*$' 
+    line: "^OPTIONS='--selinux-enabled --insecure-registry 172.30.0.0/16'"
+
+- name: "Check for existing Docker Storage device"
+  command: pvs
+  register: pvs
+
+- name: "Set Docker Storage fact if already configured"
+  set_fact:
+    docker_storage_setup: true
+  when: pvs.stdout | search('{{ docker_storage_block_device }}.*{{ docker_storage_volume_group }}')
+
+- name: "Configure Docker Storage Setup"
+  template:
+    src: docker-storage-setup.j2
+    dest: /etc/sysconfig/docker-storage-setup
+  when: docker_storage_setup is undefined
+  
+- name: "Run Docker Storage Setup"
+  command: docker-storage-setup
```

Comment:
docker-storage-setup shouldn't need to be explicitly called, since it is a dependency of the docker package. Also, if docker-storage-setup was already run before defining the block device and/or volume group, this step will fail.